server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -636,69 +727,57 @@ private void parseValueAddFields(ParseContext context, String value, String fiel
 
         if (fieldType.indexOptions() != IndexOptions.NONE || fieldType.stored()) {
             // convert to utf8 only once before feeding postings/dv/stored fields
-
             final BytesRef binaryValue = new BytesRef(fieldType().name() + DOT_SYMBOL + value);
-            Field field = new FlatObjectField(fieldType().name(), binaryValue, fieldType);
 
             if (fieldType().hasDocValues() == false && fieldType.omitNorms()) {
                 createFieldNamesField(context);
             }
```

Comment:
I just realized that the `omitNorms()` part of this doesn't make a lot of sense. 

Do we ever have norms for flat object fields? We probably shouldn't. I don't think we can rationally do relevance calculations over a flat object field. I think this was a bad copy-paste from `KeywordFieldMapper` that we missed.

Reply:
it 's good idea, I will remove the `omitNorms`.